service-api/module/Application/src/DWP/DwpApi/DwpApiService.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\DWP\DwpApi;
+
+use Application\DWP\AuthApi\AuthApiException;
+use Application\DWP\AuthApi\AuthApiService;
+use Application\DWP\DwpApi\DTO\CitizenRequestDTO;
+use Application\DWP\DwpApi\DTO\CitizenResponseDTO;
+use Application\DWP\DwpApi\DTO\DetailsRequestDTO;
+use Application\DWP\DwpApi\DTO\DetailsResponseDTO;
+use Application\DWP\DwpApi\DwpApiException;
+use Application\Model\Entity\CaseData;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\GuzzleException;
+use Laminas\Http\Response;
+use Psr\Log\LoggerInterface;
+use Ramsey\Uuid\Uuid;
+use Application\Aws\Secrets\AwsSecret;
+
+use function Amp\Promise\all;
+
+class DwpApiService
+{
+    private int $authCount = 0;
+    public function __construct(
+        private Client $guzzleClient,
+        private AuthApiService $authApiService,
+        private LoggerInterface $logger,
+        private string $detailsPath,
+        private string $matchPath,
+    ) {
+    }
+
+    /**
+     * @throws GuzzleException
+     * @throws AuthApiException
+     */
+    public function makeHeaders(): array
+    {
+        return [
+            'Content-Type' => 'application/json',
+            'Context' => 'application/process',
+            'Authorization' => sprintf(
+                'Bearer %s',
+                $this->authApiService->retrieveCachedTokenResponse()
+            ),
+            'Correlation-Id' => '',
+            'Policy-Id' => '',
+            'Instigating-User-Id' => ''
+        ];
+    }
+
+    public function validateNino(CaseData $caseData): array
```

Comment:
Rather than returning an unstructured array, could this just return a boolean of whether it was matched or not?